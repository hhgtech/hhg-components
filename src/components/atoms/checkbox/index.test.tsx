import 'jest-styled-components'

import React from 'react'

import { cleanup, render } from '@testing-library/react'

import { Checkbox } from '.'
import MockTheme from './../../../mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const dropdown = render(
    <MockTheme>
      <Checkbox
        name="checkbox1"
        checked={true}
        isDisabled={true}
        label="option1"
        value="1"
      />
    </MockTheme>,
  )

  const self = dropdown.container.firstChild

  expect(self).toMatchSnapshot()
  expect(self).toContainHTML('for="checkbox1-1"')
  expect(self).toContainHTML('<input')
  expect(self).toHaveAttribute('data-disabled', 'true')
})
