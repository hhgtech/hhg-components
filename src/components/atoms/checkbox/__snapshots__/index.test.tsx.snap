// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c0 input {
  width: 20px;
  height: 20px;
  border: solid 1px #737373;
  border-radius: 2px;
  -webkit-transition: background 0.4s,border-color 0.4s,box-shadow 0.4s;
  transition: background 0.4s,border-color 0.4s,box-shadow 0.4s;
  cursor: inherit;
}

.c0 input:hover,
.c0 input:active {
  border-color: #2d87f3;
  box-shadow: 0 0 2px 2px #91caff;
}

.c0 input:checked {
  border: 0;
  background: #2d87f3 url("data:image/svg+xml,%3Csvg width='12' height='9' viewBox='0 0 12 9' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M11 1L4.125 8L1 4.81819' stroke='white' stroke-width='2' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E%0A") no-repeat center center;
}

.c0 label {
  max-width: 15rem;
  margin-left: 8px;
  font-weight: 400;
  width: calc(100% - 24px);
  cursor: inherit;
}

.c0[data-disabled] {
  color: #bfbfbf;
  cursor: not-allowed;
}

.c0[data-disabled] input {
  border-color: #bfbfbf;
}

.c0[data-disabled] input:hover {
  border-color: #bfbfbf;
  box-shadow: none;
}

@media (max-width:767px) {
  .c0 label {
    font-size: 14px;
    line-height: 24px;
  }
}

@media (min-width:768px) {
  .c0 label {
    font-size: 16px;
    line-height: 22px;
  }
}

<div
  class="c0"
  data-disabled="true"
>
  <input
    checked=""
    disabled=""
    id="checkbox1-1"
    name="checkbox1"
    type="checkbox"
    value="1"
  />
  <label
    for="checkbox1-1"
  >
    option1
  </label>
</div>
`;
